src/Platforms/SYCL/syclBLAS.cpp

```diff
@@ -182,6 +182,109 @@ template sycl::event gemm(sycl::queue& handle,
                           const int ldc,
                           const std::vector<sycl::event>& events);
 
+//transpose
+template<typename T1, typename T2>
+sycl::event transpose(sycl::queue& q,
+                      const T1* restrict in,
+                      int m,
+                      int lda,
+                      T2* restrict out,
+                      int n,
+                      int ldb,
+                      const std::vector<sycl::event>& events)
+{
+  constexpr size_t tile_size = 16;
+  const size_t m_max         = ((m + tile_size - 1) / tile_size) * tile_size;
+  const size_t n_max         = ((n + tile_size - 1) / tile_size) * tile_size;
+
+  return q.submit([&](sycl::handler& cgh) {
```

Comment:
Could you remove all handler and use the compact form of q.parallel_for with events.